### PR TITLE
KAA-369: Update java sdk event api.

### PR DESCRIPTION
**Pull request by shpigor**

### CODE REVIEW

**Comment by ashvayka** [comment]
please update PR name to match our convention.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java [hunk 1]
```diff
@@ -474,4 +491,49 @@ public String getEndpointKeyHash() {
     public PrivateKey getClientPrivateKey() {
         return kaaClientState.getPrivateKey();
     }
+
+    @Override
+    public String regenerateEndpointAccessToken() {
+        return null;
```
Comment (ashvayka) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/AbstractKaaClient.java]: Any comments? :-)

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/GenericKaaClient.java [hunk 2]
```diff
@@ -497,4 +500,88 @@
...
+     * @see org.kaaproject.kaa.client.event.EndpointKeyHash
+     * @see OnDetachEndpointOperationCallback
+     */
+    void detachEndpoint(EndpointKeyHash endpointKeyHash, OnDetachEndpointOperationCallback resultListener);
+
+    /**
+     * Creates user attach request using default verifier. Default verifier is selected during SDK generation.
+     * If there was no default verifier selected this method will throw runtime exception.
+     *
+     * @param userExternalId
```
Comment (ghost) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/GenericKaaClient.java:543]: I think we have to add documentation for these parameters (both attachUser versions).

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/EventTransport.java [hunk 3]
```diff
@@ -64,7 +60,14 @@
      */
     void setEventManager(EventManager manager);
 
+    /**
```
Comment (ghost) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/channel/EventTransport.java:63]: Please, get rid off these empty comment blocks or add some doc if it is useful there.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/DefaultEventManager.java [hunk 4]
```diff
@@ -118,12 +117,12 @@ public void clearState() {
...
-    public void produceEvent(String eventFqn, byte[] data, String target) throws IOException {
+    public void produceEvent(String eventFqn, byte[] data, String target) {
         produceEvent(eventFqn, data, target, null);
     }
 
     @Override
-    public void produceEvent(String eventFqn, byte[] data, String target, TransactionId trxId) throws IOException {
+    public void produceEvent(String eventFqn, byte[] data, String target, TransactionId trxId) {
         if (trxId == null) {
             LOG.info("Producing event [eventClassFQN: {}, target: {}]"
```
Comment (ghost) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/DefaultEventManager.java:127]: Please, move the log level up to DEBUG or higher.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/registration/DefaultEndpointRegistrationManager.java [hunk 5]
```diff
@@ -207,45 +171,31 @@ public void onUpdate(List<EndpointAttachResponse> attachResponses,
...
-            resultListener.sendResponse("endpoint attached", result, keyHash);
-        }
-    }
-
-    private void notifyDetachedListener(org.kaaproject.kaa.common.endpoint.gen.SyncResponseResultType result,
-            EndpointOperationResultListener resultListener) {
-        if(resultListener != null){
-            resultListener.sendResponse("endpoint detached", result, null);
+    private void notifyAttachedListener(SyncResponseResultType result, OnAttachEndpointOperationCallback operationCallback, EndpointKeyHash keyHash) {
+        if(operationCallback != null){
```
Comment (ghost) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/registration/DefaultEndpointRegistrationManager.java:191]: Please, add code-style spaces.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/registration/DefaultEndpointRegistrationManager.java [hunk 6]
```diff
@@ -207,45 +171,31 @@ public void onUpdate(List<EndpointAttachResponse> attachResponses,
...
         }
     }
 
-    private void notifyListeners() {
-        synchronized (attachedListChangedListeners) {
-            for (AttachedEndpointListChangedListener listener : attachedListChangedListeners) {
-                listener.onAttachedEndpointListChanged(getAttachedEndpointList());
-            }
+    private void notifyDetachedListener(SyncResponseResultType result, OnDetachEndpointOperationCallback operationCallback) {
+        if(operationCallback != null){
```
Comment (ghost) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/registration/DefaultEndpointRegistrationManager.java:197]: Please, add code-style spaces.

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/registration/OnDetachEndpointOperationCallback.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2015 CyberVision, Inc.
```
Comment (ghost) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/event/registration/OnDetachEndpointOperationCallback.java:2]: 2014-2015

File: client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java [hunk 8]
```diff
@@ -257,6 +258,13 @@ public void persist() {
     }
 
     @Override
+    public String refreshEndpointAccessToken() {
+        String newAccessToken = UUID.randomUUID().toString();
+        setEndpointAccessToken(newAccessToken);
+        return newAccessToken;
```
Comment (ghost) [inline client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/persistence/KaaClientPropertiesState.java:264]: Maybe it is redundant, but I suggest to get the access token the same way as we set it. That is through getEndpointAccessToken.

File: client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/event/registration/DefaultEndpointRegistrationManagerTest.java [hunk 9]
```diff
@@ -147,6 +147,7 @@ public void checkEndpointAttachDetachResponse() throws Exception {
         }
     }
 
+    @Ignore("Ignore during removing endpoint list change listener .")
```
Comment (ghost) [inline client/client-multi/client-java-core/src/test/java/org/kaaproject/kaa/client/event/registration/DefaultEndpointRegistrationManagerTest.java:150]: So must this test be re-written, mustn't it?

File: examples/smarthousedemo/source/src/org/kaaproject/kaa/demo/smarthousedemo/command/AttachEndpointCommand.java [hunk 10]
```diff
@@ -39,20 +41,18 @@ public AttachEndpointCommand(KaaClient client, String endpointAccessToken) {
...
-                    }
-                    else {
-                        Log.e("Kaa", "Unable to attach endpoint to user account!");
-                        onException(new CommandException("Unable to attach endpoint to user account!"));
-                    }
+        client.attachEndpoint(accessToken, new OnAttachEndpointOperationCallback() {
+            @Override
+            public void onAttach(SyncResponseResultType result, EndpointKeyHash context) {
+                if (result == SyncResponseResultType.SUCCESS) {
+                    Log.d("Kaa", "Endpoint attached to user account!");
```
Comment (ghost) [inline examples/smarthousedemo/source/src/org/kaaproject/kaa/demo/smarthousedemo/command/AttachEndpointCommand.java:48]: Please, move the log tag to the named constant.

File: examples/smarthousedemo/source/src/org/kaaproject/kaa/demo/smarthousedemo/command/DetachEndpointCommand.java [hunk 11]
```diff
@@ -37,19 +37,17 @@ public DetachEndpointCommand(KaaClient client, String endpointKeyHash) {
...
-                    }
-                    else {
-                        Log.w("Kaa", "Endpoint already detached from user account!");
-                        onComplete(false);
-                    }
+        client.detachEndpoint(endpointKey, new OnDetachEndpointOperationCallback() {
+            @Override
+            public void onDetach(SyncResponseResultType result) {
+                if (result == SyncResponseResultType.SUCCESS) {
+                    Log.d("Kaa", "Endpoint detached from user account!");
```
Comment (ghost) [inline examples/smarthousedemo/source/src/org/kaaproject/kaa/demo/smarthousedemo/command/DetachEndpointCommand.java:44]: Please, move the log tag to the named constant.

**Pull request merged**